clients/drcachesim/analyzer_multi.cpp

```diff
@@ -60,13 +62,50 @@
 #include "tools/reuse_distance_create.h"
 #include "tools/reuse_time_create.h"
 #include "tools/view_create.h"
+#include "tools/loader/external_config_file.h"
+#include "tools/loader/external_tool_creator.h"
 
 namespace dynamorio {
 namespace drmemtrace {
 
 using ::dynamorio::droption::droption_parser_t;
 using ::dynamorio::droption::DROPTION_SCOPE_ALL;
 
+analysis_tool_t *
+analyzer_multi_t::create_external_tool(const std::string &id)
+{
+    analysis_tool_t *tool = nullptr;
+
+    std::string tools_dir(op_dr_root.get_value());
+    tools_dir.append("/tools/");
+    directory_iterator_t end;
+    directory_iterator_t iter(tools_dir);
+
+    if (!iter) {
+        return nullptr;
+    }
+
+    for (; iter != end; ++iter) {
+        if ((*iter).find(".drcachesim") != std::string::npos) {
+            std::string abs_path(tools_dir);
+            abs_path.append(*iter);
+            external_tool_config_file_t config(op_dr_root.get_value(), abs_path);
+            if (config.valid_ && config.id_ == id) {
+                try {
+                    external_tool_creator creator(config.creator_path_.c_str());
+                    DR_ASSERT(creator.get_id_() != id.c_str());
+                    tool = creator.create_tool_();
```

Comment:
See comment below: this is currently not guaranteed to not be nullptr; if not checked elsewhere, should check here.

Reply:
There is checking for nullptr after this function.

Reply:
OK.